src/main/java/com/intuit/graphql/orchestrator/schema/transform/RequireTransformer.java

```diff
@@ -0,0 +1,49 @@
+package com.intuit.graphql.orchestrator.schema.transform;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.intuit.graphql.graphQL.Directive;
+import com.intuit.graphql.graphQL.FieldDefinition;
+import com.intuit.graphql.graphQL.InterfaceTypeDefinition;
+import com.intuit.graphql.graphQL.ObjectTypeDefinition;
+import com.intuit.graphql.graphQL.TypeDefinition;
+import com.intuit.graphql.orchestrator.federation.requiresdirective.RequireValidator;
+import com.intuit.graphql.orchestrator.xtext.XtextGraph;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static com.intuit.graphql.orchestrator.utils.FederationUtils.FEDERATION_REQUIRES_DIRECTIVE;
+import static com.intuit.graphql.orchestrator.utils.XtextTypeUtils.getFieldDefinitions;
+
+public class RequireTransformer implements Transformer<XtextGraph, XtextGraph> {
+
+    @VisibleForTesting
+    RequireValidator requireValidator = new RequireValidator();
+
+    @Override
+    public XtextGraph transform(final XtextGraph source) {
+
+        if(source.getServiceProvider().isFederationProvider()) {
+            for(final TypeDefinition typeDefinition : source.getTypes().values()) {
```

Comment:
try using streams here

Reply:
I was suppose to look at that before I push; thanks for the reminder!